functorch/csrc/DynamicLayer.cpp

```diff
@@ -513,6 +541,20 @@ void dynamicLayerFrontFallback(const c10::OperatorHandle& op, torch::jit::Stack*
 
   // Re-dispatch
   op.callBoxed(stack);
+  // checking exclude set because we should only do this if we actually hit a functionalization kernel.
+  if (layer.key() == DispatchKey::Functionalize && !exclude.has(DispatchKey::Functionalize)) {
```

Comment:
Checking `!exclude.has(DispatchKey::Functionalize)` is a little fishy here, I need to think a bit more about it. I'm worried that in callBoxed if we have nested functionalize transforms then some of them might reset the exclude? But we don't support nested functionalize transforms yet so this is probably fine for now...